lib/rex/ui/text/shell.rb

```diff
@@ -149,7 +149,13 @@ def run(&block)
         if input.eof? || line == nil
           self.stop_count += 1
           next if self.stop_count > 1
-          run_single('quit')
+
+          if block
+            block.call('quit')
+          elsif respond_to?(:run_single)
+            # PseudoShell does not provide run_single
+            run_single('quit')
+          end
```

Comment:
Do we have a way to handle this if no block was passed and the code does not respond to `run_single`? Will this not end up falling through to the  following code which will also end up calling `run_single` without appropriate checks?

Reply:
Well if there's no block to run commands and there's no `#run_single` method to run commands, I'm not sure it's necessary or possible to run the `quit` command. In this case, falling through seems reasonable. If there's someplace where falling through without running the `quit` command seems wrong, let me know.